novawallet/Modules/BackupMnemonicCard/BackupMnemonicCardViewController.swift

```diff
@@ -0,0 +1,258 @@
+        at indexPath: IndexPath
+    ) -> UICollectionReusableView {
+        let view = collectionView.dequeueReusableSupplementaryViewWithType(
+            TitleCollectionHeaderView.self,
+            forSupplementaryViewOfKind: kind,
+            for: indexPath
+        )!
+        setupCollectionHeader(view)
+
+        return view
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout _: UICollectionViewLayout,
+        referenceSizeForHeaderInSection _: Int
+    ) -> CGSize {
+        CGSize(width: collectionView.bounds.width, height: UIConstants.headerHeight)
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCellWithType(Cell.self, for: indexPath)!
+        cell.view.view.attributedText = createIndexedWordAttributedString(
+            for: words[indexPath.item],
+            index: indexPath.item
+        )
+
+        return cell
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAt _: IndexPath
+    ) -> CGSize {
+        guard let flowLayout = collectionViewLayout as? UICollectionViewFlowLayout else { return CGSize.zero }
+
+        let totalSpace = flowLayout.sectionInset.left
+            + flowLayout.sectionInset.right
+            + (flowLayout.minimumInteritemSpacing * CGFloat(UIConstants.itemsPerRow - 1))
+
+        let width = Int((collectionView.bounds.width - totalSpace) / CGFloat(UIConstants.itemsPerRow))
+
+        return CGSize(width: CGFloat(width), height: CGFloat(UIConstants.wordHeight))
+    }
+}
+
+// MARK: BackupMnemonicCardViewLayoutDelegate
+
+extension BackupMnemonicCardViewController: BackupMnemonicCardViewLayoutDelegate {
+    func didTapCardCover() {
+        presenter.mnemonicCardTapped()
+    }
+}
+
+// MARK: Private
+
+private extension BackupMnemonicCardViewController {
+    func setupView() {
+        rootView.collectionView.dataSource = self
+        rootView.collectionView.delegate = self
+        rootView.delegate = self
+    }
+
+    func setupNavigationBarTitle(with viewModel: ViewModel) {
+        let iconDetailsView: IconDetailsView = .create(with: { view in
+            view.iconWidth = UIConstants.walletIconSize.width
+
+            viewModel.walletIcon?.loadImage(
+                on: view.imageView,
+                targetSize: UIConstants.walletIconSize,
+                animated: true
+            )
+            view.detailsLabel.apply(style: .semiboldBodyPrimary)
+            view.detailsLabel.text = viewModel.walletName
+        })
+
+        navigationItem.titleView = iconDetailsView
+    }
+
+    func createIndexedWordAttributedString(for word: String, index: Int) -> NSAttributedString {
+        let buttonTitleStr = NSMutableAttributedString(
```

Comment:
Probably, NSAttributedString.coloredItems approach looks more readable and compact